packages/govern-core/contracts/Govern.sol

```diff
@@ -23,18 +27,22 @@ contract Govern is ERC3000Executor, ACL {
         emit ETHDeposited(msg.sender, msg.value);
     }
 
-    function exec(ERC3000Data.Action[] memory actions) override public auth(EXEC_ROLE) returns (bytes[] memory) {
+    function exec(ERC3000Data.Action[] memory actions, bytes32 allowFailuresMap, bytes32 memo) override public auth(EXEC_ROLE) returns (bytes[] memory) {
+        require(actions.length <= MAX_ACTIONS, "govern: too many"); // need to limit since we use 256-bit bitmaps
+        
         bytes[] memory execResults = new bytes[](actions.length);
+        bytes32 failureMap = BitmapLib.empty; // start with an empty bitmap
 
-        for (uint256 i = 0; i < actions.length; i++) {
-            // todo: optimize with assembly
+        for (uint8 i = 0; i < actions.length; i++) { // can use uint8 given the action limit
+            // TODO: optimize with assembly
```

Comment:
Should we do this anytime soon? 😄

Reply:
Unsure if it is worth it, we should do some gas profiling when it is time for fine polishing, we are not there just yet